frontend/simple/views/containers/Proposals.vue

```diff
@@ -55,138 +37,140 @@
-            title: 'Min Income',
-            text: 'You had proposed to change <strong>mincome from $200 to $150</strong>',
-            members: 7,
-            votes: 0,
-            ownProposal: true
-          }
-        }
-      ]
-    }
+    Voting
   },
   computed: {
-    groupProposals () {
+    ...mapGetters([
+      'currentUserIdentityContract',
+      'currentGroupState'
+    ]),
+    proposalList () {
       const notVoted = []
       const alreadyVoted = []
       const own = []
 
-      for (var i = 0, l = this.mockProposals.length; i < l; i++) {
-        const proposal = this.mockProposals[i]
-        if (proposal.data.ownProposal) {
-          own.push(proposal)
-        } else if (proposal.data.userVote === null) {
-          notVoted.push(proposal)
-        } else if (proposal.data.userVote !== null) {
-          alreadyVoted.push(proposal)
+      const groupData = this.currentGroupState
+      const proposals = groupData.proposals
+      const userData = this.currentUserIdentityContract.attributes
+
+      for (let hash in proposals) {
+        const proposal = proposals[hash]
+        const proposalData = {
+          type: proposal.type, // 'invitation' or 'removal' for member, field name for rule/mincome
+          votes: {
+            total: Object.entries(groupData.profiles).length,
+            received: proposal.for.length + proposal.against.length - 1 // initiator's for vote doesn't count here
+          },
+          value: proposal.value || proposal.candidate || null,
+          originalValue: groupData[proposal.type] || null,
+          ownVote: proposal.for.includes(userData.name) || proposal.against.includes(userData.name)
+            ? proposal.for.includes(userData.name)
+            : null,
+          isOwnProposal: proposal.initiator === userData.name,
+          initiator: proposal.initiator,
+          hash: hash
+        }
+
+        if (proposalData.isOwnProposal) {
+          own.push(proposalData)
+        } else if (proposalData.ownVote === null) {
+          notVoted.push(proposalData)
+        } else {
+          alreadyVoted.push(proposalData)
         }
       }
 
       return { notVoted, alreadyVoted, own }
     },
     allVoted () {
-      const { notVoted, alreadyVoted } = this.groupProposals
+      const { notVoted, alreadyVoted } = this.proposalList
 
       return alreadyVoted.length && notVoted.length === 0
     }
   },
   methods: {
-    handleVoteAgainst () {
-      console.log('TODO Logic - The user voted against')
+    async handleVoteAgainst (hash) {
+      try {
+        // Create a vote against the proposal
+        const groupId = this.$store.state.currentGroupId
+        const latest = await backend.latestHash(groupId)
+        const vote = new Events.HashableGroupVoteAgainstProposal({
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash
+        }, latest)
+        await backend.publishLogEntry(groupId, vote)
```

Comment:
So, recall that some of the changes I'm going to do will mean that `Events.HashableGroupVoteAgainstProposal` might not exist. It would help therefore if you continued as before and moved these `new Events.*` stuff into SBP selectors defined together in another file, and then call the SBP selector here with the necessary parameters gathered from the UI. That will make it simpler for me to change them all at once if necessary by editing just a single file instead of sending in a PR with a whole bunch of .vue files changed.

Reply:
@hubudibu ignore that ☝️ suggestion!

As I'm working on my PR I've realized for now it's probably best to leave that as-is!